### PR TITLE
springExerciseNR1

### PR DESCRIPTION
**Pull request by Kalixone**

### CODE REVIEW

File: src/main/java/mate/academy/springbootintro/config/SecurityConfig.java [hunk 1]
```diff
@@ -39,7 +38,7 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .csrf(AbstractHttpConfigurer::disable)
                 .authorizeHttpRequests(
                         auth -> auth
-                                .requestMatchers(HttpMethod.POST, "/api/auth/**")
+                                .requestMatchers("api/auth/**")
```
Comment (nowickimj) [inline src/main/java/mate/academy/springbootintro/config/SecurityConfig.java]: I think that you're missing a swagger path here

File: src/main/java/mate/academy/springbootintro/repository/book/BookRepository.java [hunk 2]
```diff
@@ -3,8 +3,14 @@
...
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
+import java.util.List;
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+
+    @Query("SELECT b FROM Book b LEFT JOIN FETCH b.categories c"
```
Comment (nowickimj) [inline src/main/java/mate/academy/springbootintro/repository/book/BookRepository.java]: No need for nested SELECT - try `SELECT b FROM Book b JOIN FETCH b.categories c WHERE c.id = :categoryId"`

File: src/main/java/mate/academy/springbootintro/service/CategoryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto createCategoryRequestDto) {
+        Category entity = categoryMapper.toEntity(createCategoryRequestDto);
+        return categoryMapper.toDto(categoryRepository.save(entity));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryDto categoryDto) {
+        Category existingCategory = categoryRepository.findById(id)
```
Comment (nowickimj) [inline src/main/java/mate/academy/springbootintro/service/CategoryServiceImpl.java]: You're duplicating the call from `getById` method - extract this code to a private method.

**Review by nowickimj — CHANGES_REQUESTED** [review]
Few remarks

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -6,4 +6,10 @@ databaseChangeLog:
...
   - include:
-      file: db/changelog/changes/04-create-user-roles-table.yaml
\ No newline at end of file
+      file: db/changelog/changes/04-create-user-roles-table.yaml
+  - include:
+      file: db/changelog/changes/05-create-categories-table.yaml
+  - include:
+      file: db/changelog/changes/06-insert-categories.yaml
+  - include:
+      file: db/changelog/changes/07-create-book-categories-table.yaml
```
Comment (okuzan) [inline src/main/resources/db/changelog/db.changelog-master.yaml]:
```suggestion
      file: db/changelog/changes/07-create-book-categories-table.yaml
      
```
always leave last line empty

File: src/main/resources/db/changelog/changes/07-create-book-categories-table.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+                    foreignKeyName: fk_book
+                    references: books(id)
+              - column:
+                  name: category_id
+                  type: bigint
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+                    foreignKeyName: fk_category
+                    references: categories(id)
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/07-create-book-categories-table.yaml]:
```suggestion
                    references: categories(id)
                    
```

File: src/main/resources/db/changelog/changes/06-insert-categories.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+            columns:
+              - column:
+                  name: id
+                  value: 2
+              - column:
+                  name: name
+                  value: "action"
+              - column:
+                  name: description
+                  value: "pif-paf"
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/06-insert-categories.yaml]:
```suggestion
                  value: "pif-paf"
                  
```

File: src/main/resources/db/changelog/changes/05-create-categories-table.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+                  type: VARCHAR(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: description
+                  type: VARCHAR(1000)
+              - column:
+                  name: isDeleted
+                  type: BOOLEAN
+                  defaultValue: false
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/05-create-categories-table.yaml]:
```suggestion
                  defaultValue: false
                  
```

File: src/main/java/mate/academy/springbootintro/service/BookService.java [hunk 8]
```diff
@@ -19,5 +20,8 @@ public interface BookService {
     BookDto updateBook(Long id, UpdateBookRequestDto updateBookRequestDto);
 
     List<BookDto> search(BookSearchParameters params);
+
+    List<BookDtoWithoutCategoryIds> findBooksByCategoryId(Long id);
 }
 
+
```
Comment (okuzan) [inline src/main/java/mate/academy/springbootintro/service/BookService.java]:
```suggestion
```

File: src/main/java/mate/academy/springbootintro/model/Category.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.GenerationType;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Data
+@Table(name = "categories")
```
Comment (okuzan) [inline src/main/java/mate/academy/springbootintro/model/Category.java]:
```suggestion
@Getter
@Setter
@SQLDelete(sql = "UPDATE categories SET is_deleted = TRUE WHERE id = ?")
@SQLRestriction(clause = "is_deleted = FALSE")
@Table(name = "categories")
```
using Data on entities is generally discouraged, Where is now deprecated

File: et --hard HEAD~1q [hunk 10]
```diff
@@ -0,0 +1,95 @@
+[33mcommit 88670d94263b30e5792852275db7a9b7893fddc8[m[33m ([m[1;36mHEAD -> [m[1;32mmain[m[33m, [m[1;31morigin/springBootExerciseNr1[m[33m, [m[1;31morigin/main[m[33m, [m[1;32mspringBootExerciseNr1[m[33m, [m[1;32msecurityexercise[m[33m)[m
```
Comment (okuzan) [inline et --hard HEAD~1q:1]: this file probably got committed accidentally?

**Review by okuzan — CHANGES_REQUESTED** [review]

File: et --hard HEAD~1q (hunk 10 above)
Comment (Kalixone) [reply et --hard HEAD~1q:1]: yes, it was

**Pull request merged**